components/ILIAS/User/src/Search/EndpointConfigurator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\User\Search;
+
+interface EndpointConfigurator
+{
+    public function getParentClassPath(): array;
+    public function getAdditionalAnswerElements(
```

Comment:
Shouldn't we use `array` (with an additional `phpdoc` comment describing the type/shape of this array) as a return type here?

I guess this interface can be used for consumers (like `Mail`) to add further providers, which may add additional "Autocomplete Items" to the search response.
If I understood this correctly, I suggest adding a short description here.